### ISSUE TITLE
What is the difference between rotmat and global_xform?

### ISSUE THREAD

**Issue by xwg0508** [c-he/NeMF#18]
Hi , thank you for your code ! I want to ask the difference between rotmat(or rot6d) and global_xform?
From the code in **amass.py**  ,  i see the global_xform was gotten by the following code:
<img width="386" alt="image" src="https://github.com/c-he/NeMF/assets/130576216/d72134fd-410e-432c-b587-75ccabb2f5bd">
<img width="546" alt="image" src="https://github.com/c-he/NeMF/assets/130576216/31017e48-0c3b-43f2-8b3b-ffe5278bb26b">
SO What is the difference between rot6d and global_xform?Why we get global_xform through rot6d
AND in training and applying , the global_xform was seen as rotmat, such as the following code:
<img width="665" alt="image" src="https://github.com/c-he/NeMF/assets/130576216/1c801be4-39ee-4202-937d-5425aa097841">
SO WHAT is the global_xform?

**Comment by c-he** [comment]
`rot6d` and `rotmat` are joint rotations relative to their parent joint, while `global_xform` is the rotation relative to the root joint. In training, I used `global_xform` since this global rotation representation performed better in our experiments. I should have given it a different name in the training code, so sorry for the confusion caused by my poor coding style :)

**Comment by ojss** [comment]
@c-he something is confusing me here. The model (generative) returns `local_motion` but that is again passed through `global_to_local` of the `fk` layer to get `local_rotmat` but this doesn't get used in the loss [here](https://github.com/c-he/NeMF/blob/6c9810105f0b9019b3b0b255611c9d5bbbf1b7a4/src/nemf/generative.py#L278).

The loss is between the 6d `local_motion` and input `rotmat` which is still the local (relative to parent) representation the [preprocessing code](https://github.com/c-he/NeMF/blob/6c9810105f0b9019b3b0b255611c9d5bbbf1b7a4/src/datasets/amass.py#L103). 

I'm not sure what I'm missing 😦

**Comment by c-he** [comment]
Basically we need to clarify the variable names first:
- `global_motion` and `local_motion`: they refer to the output of our motion field, where `global` and `local` are used to distinguish between the rotation of the root joint **(global)** and the rotations of other joints **(local)**.
- `global_to_local()` in `fk`: in this context, global and local distinguish between rotations relative to the root joint **(global)** and those relative to the parent joint **(local)**. To compute FK, we first need to convert the decoded rotations (`rotmat_recon`), which are relative to the root joint, into rotation matrices relative to their respective parent joints -- denoted as `local_rotmat`. Using these local rotation matrices, we can then apply the chain rule to compute joint positions.
https://github.com/c-he/NeMF/blob/6c9810105f0b9019b3b0b255611c9d5bbbf1b7a4/src/nemf/generative.py#L191-L207

During training, the geodesic loss is actually computed between rotation matrices relative to the root joint, see:
https://github.com/c-he/NeMF/blob/6c9810105f0b9019b3b0b255611c9d5bbbf1b7a4/src/nemf/generative.py#L102

Again, I'm sorry for the confusion caused by my poor naming style. Please let me know if you have any further questions.

**Comment by ojss** [comment]
@c-he ah I get it now thank you! I missed Line 102, this makes so much more sense now haha. Once again, appreciate your quick replies :)
